lib/moonshot/build_mechanism/github_release.rb

```diff
@@ -140,6 +143,20 @@ def hub_release_exists(semver)
       false
     end
 
+    def hub_find_remote_tag(tag_name)
+      attempts = 0
+      # Attempt to find the build. Rescue and re-attempt if the build can not
+      # be found on github yet.
+      begin
+        sleep 1 # Give some time for tag to showup.
+        sh_out("hub ls-remote --exit-code --tags upstream #{tag_name}")
+      rescue RuntimeError => e
+        attempts += 1
+        retry unless attempts >= MAX_GITHUB_TAG_FIND_ATTEMPTS
```

Comment:
This should probably be configurable and we may want to sleep for longer than a second. On multiple occasions I've seen GitHub take more than a minute to recognize something that's been pushed to it.